packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnSpanning.ts

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridColumnIndex, GridCellMeta } from '../../../models/gridColumnSpanning';
+import { GridRowId } from '../../../models/gridRows';
+import { GridColumnSpanning } from '../../../models/api/gridColumnSpanning';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+import { GridEvents } from '../../../models/events/gridEvents';
+import { GridCellParams } from '../../../models/params/gridCellParams';
+import { GridStateColDef } from '../../../models/colDef/gridColDef';
+import { GridApiCommunity } from '../../../models/api/gridApiCommunity';
+
+/**
+ * @requires useGridColumns (method, event)
+ * @requires useGridParamsApi (method)
+ */
+export const useGridColumnSpanning = (apiRef: React.MutableRefObject<GridApiCommunity>) => {
+  const lookup = React.useRef<Record<GridRowId, Record<GridColumnIndex, GridCellMeta>>>({});
+
+  const setCellMeta = React.useCallback(
+    (rowIndex: GridRowId, columnIndex: GridColumnIndex, size: GridCellMeta) => {
+      const sizes = lookup.current;
+      if (!sizes[rowIndex]) {
+        sizes[rowIndex] = {};
+      }
+
+      sizes[rowIndex][columnIndex] = size;
+    },
+    [],
+  );
+
+  const getCellMeta = React.useCallback<GridColumnSpanning['unstable_getCellMeta']>(
```

Comment:
Maybe avoid the term `meta` which is not self describing.

For the columns we deprecated the meta method and selector and replace it by `gridColumnPositionsSelector` and `gridColumnTotalWidthSelector`.

Reply:
Maybe `getCellColSpan` or `getCellColSpanInfo`?

Reply:
Will we want to put info about row span in this method when #207 will be implemented ?

`getCellSpanning` ?
`getCellSpan` ?

Reply:
I'm not sure about that. I would aim at separating `colSpan` and `rowSpan`, but I didn't explore rowSpan and I'm not sure it's 100% possible.
Either way, those methods are internal and we can always rename them.
I'm leaning towards using `getCellColSpanInfo` for now. What do you think?

Reply:
> Either way, those methods are internal and we can always rename them.

True, go for `getCellColSpanInfo` then, it's explicit even if not very sexy so fine for an internal method.